core/ledger/pvtdatastorage/reconcile_missing_pvtdata_test.go

```diff
@@ -7,196 +7,569 @@ SPDX-License-Identifier: Apache-2.0
+	require.NoError(t, store.Commit(1, blocksPvtData[1].pvtData, blocksPvtData[1].missingDataInfo))
+	require.NoError(t, store.Commit(2, blocksPvtData[2].pvtData, blocksPvtData[2].missingDataInfo))
+
+	assertMissingDataInfo(t, store, missingDataSummary, 2)
+
+	// COMMIT some of the missing data in the block 1 and block 2
+	oldBlockTxPvtDataInfo := []*blockTxPvtDataInfoForTest{
+		{
+			blkNum: 1,
+			txNum:  1,
+			pvtDataPresent: map[string][]string{
+				"ns-1": {"coll-1"},
+				"ns-2": {"coll-1"},
+			},
+			pvtDataMissing: map[string][]string{
+				"ns-1": {"coll-2"},
+				"ns-2": {"coll-2"},
+			},
+		},
+		{
+			blkNum: 1,
+			txNum:  2,
+			pvtDataPresent: map[string][]string{
+				"ns-1": {"coll-1"},
+			},
+			pvtDataMissing: map[string][]string{
+				"ns-1": {"coll-2"},
+			},
+		},
+		{
+			blkNum: 2,
+			txNum:  1,
+			pvtDataMissing: map[string][]string{
+				"ns-1": {"coll-1", "coll-2"},
+			},
+		},
+		{
+			blkNum: 2,
+			txNum:  3,
+			pvtDataPresent: map[string][]string{
+				"ns-1": {"coll-1"},
+			},
+		},
 	}
 
-	err = store.CommitPvtDataOfOldBlocks(oldBlocksPvtData)
-	require.NoError(t, err)
+	blocksPvtData, missingDataSummary = constructPvtDataForTest(t, oldBlockTxPvtDataInfo)
+	oldBlocksPvtData := map[uint64][]*ledger.TxPvtData{
+		1: blocksPvtData[1].pvtData,
+		2: blocksPvtData[2].pvtData,
+	}
+	require.NoError(t, store.CommitPvtDataOfOldBlocks(oldBlocksPvtData, nil))
 
-	// ENSURE THAT THE PREVIOUSLY MISSING PVTDATA OF BLOCK 1 & 2 EXIST IN THE STORE
-	ns1Coll1Blk1Tx1 := &dataKey{nsCollBlk: nsCollBlk{ns: "ns-1", coll: "coll-1", blkNum: 1}, txNum: 1}
-	ns2Coll1Blk1Tx1 := &dataKey{nsCollBlk: nsCollBlk{ns: "ns-2", coll: "coll-1", blkNum: 1}, txNum: 1}
-	ns1Coll1Blk1Tx2 := &dataKey{nsCollBlk: nsCollBlk{ns: "ns-1", coll: "coll-1", blkNum: 1}, txNum: 2}
-	ns3Coll1Blk1Tx2 := &dataKey{nsCollBlk: nsCollBlk{ns: "ns-3", coll: "coll-1", blkNum: 1}, txNum: 2}
-	ns1Coll1Blk2Tx3 := &dataKey{nsCollBlk: nsCollBlk{ns: "ns-1", coll: "coll-1", blkNum: 2}, txNum: 3}
-
-	require.True(t, testDataKeyExists(t, store, ns1Coll1Blk1Tx1))
-	require.True(t, testDataKeyExists(t, store, ns2Coll1Blk1Tx1))
-	require.True(t, testDataKeyExists(t, store, ns1Coll1Blk1Tx2))
-	require.True(t, testDataKeyExists(t, store, ns3Coll1Blk1Tx2))
-	require.True(t, testDataKeyExists(t, store, ns1Coll1Blk2Tx3))
-
-	// pvt data retrieval for block 2 should return the just committed pvtdata
-	var nilFilter ledger.PvtNsCollFilter
-	retrievedData, err := store.GetPvtDataByBlockNum(2, nilFilter)
-	require.NoError(t, err)
-	for i, data := range retrievedData {
-		require.Equal(t, data.SeqInBlock, oldBlocksPvtData[2][i].SeqInBlock)
-		require.True(t, proto.Equal(data.WriteSet, oldBlocksPvtData[2][i].WriteSet))
+	for _, b := range blocksPvtData {
+		for _, dkey := range b.dataKeys {
+			require.True(t, testDataKeyExists(t, store, dkey))
+		}
 	}
+	assertMissingDataInfo(t, store, missingDataSummary, 2)
+}
+
+func TestCommitPvtDataOfOldBlocksWithBTL(t *testing.T) {
```

Comment:
To be honest, I haven't compared this test with it's previous version because it's significantly changed in this refactoring...so, had to look a fresh but I find it very hard to track at places. Primarily because, one would have to keep track of the flow and correlate with the current state of the store while keeping in mind the different elements of sample data.  

For instance, I didn't find the coverage of purging the depriortized list when it is present before the expiry (I see the one that is only added after the expiry). Similarly, deletion of prioritized data is not covered. I feel that this test should be broken into different test cases, perhaps in the line of how you have done in the test `TestCommitPvtDataOfOldBlocksWithDeprioritization`.

Reply:
Puring is tested as part of `TestExpiryDataNotIncluded()` -- which is an existing test. This might require a refactoring but not going to do it in this PR.

Reply:
In that case, I don't understand the value of code from line 163 through 238. If I get you right, all you wanted to test was that the data supplied to the call `store.CommitPvtDataOfOldBlocks(oldBlocksPvtData, deprioritizedList)` at line 270 should be kind of a noop if the data has arrived after it's expiry. Did I miss something?

Reply:
This does present in the existing test as well. All I did was that I split `TestCommitPvtDataOfOldBlocks` into `TestCommitPvtDataOfOldBlocks` and `TestCommitPvtDataOfOldBlocksWithBTL` and refactored as it was hard to read the existing test.

Reply:
> This does present in the existing test as well
Don't get me wrong but, IMO that does not go well justifying the refactoring that has taken place. To me, this still looks unreadable and hard to get what this test tries to achieve....  It may be looking an improved version to you because you just wrote it and hence is fresh in your mind but I found it as cryptic as before.

Reply:
Agree. To me, it reads better than the previous version. I did multiple iterations and came up with this. I am out of ideas here. If you have a better way to organize this test, please feel free to push a PR.

Reply:
May be you missed in the previous comment, something that does not require drastic change... just remove the unnecessary code. 
- `setup part` - just do fake commits for a few blocks (with no private data)
- `API call` a single call to `store.CommitPvtDataOfOldBlocks(oldBlocksPvtData, deprioritizedList)`
- `Verification` - the API side-effect is noop